peekingduck/declarative_loader.py

```diff
@@ -64,6 +66,13 @@ def __init__(
     ) -> None:
         self.logger = logging.getLogger(__name__)
 
+        parent_path = str(pipeline_path.parent)
+        if parent_path != Path.cwd() and parent_path[0] == "/":
```

Comment:
Just to ensure I'm understanding the proposed changes correctly, something like
```
/root
+-PeekingDuck
+-project_1
  +-pipeline_1.yml
  \-wave.mp4
```
where 
```yaml
# pipeline_1.yml
nodes:
  - input.visual:
      source: wave.mp4
  - output.screen
```
and executing the following in `/root/PeekingDuck`
```
peekingduck run --config_path /root/project_1/pipeline_1.yml
```
should work with no errors?

I've tested the above on my ubuntu machine successfully, but I wasn't able to do so on my Windows machine with 
```
peekingduck run --config_path C:\root\project_1\pipeline_1.yml
```
Is it possible to use [`is_absolute()`](docs.python.org/3/library/pathlib.html#pathlib.PurePath.is_absolute) to check the path instead?

Reply:
The attached screenshot of the runtime messages shows this feature running in developer mode:

![image](https://user-images.githubusercontent.com/13075447/174558549-78605cdd-5a9f-4c97-8875-769489e31076.png)

Reply:
Strangely, `python . --config_path /some/pipeline/yaml/file` works on macOS:

![image](https://user-images.githubusercontent.com/13075447/174705326-393af139-b70d-48fa-968c-c48b635be6d4.png)

Reply:
https://docs.python.org/3/using/cmdline.html#using-on-interface-options 

Under the 1.1.1 > <script>, I think doing `python __main__.py` adds the directory containing `__main__.py` to `sys.path`

Reply:
Ok, I initially only brought this up because of the inconsistent behavior. But since this looks like it will require some additional implementation effort and doesn't directly affect the users, I think we can shelve it first and proceed with the rest of the PR?

Reply:
Yep, let's KIV this. I was curious about the runtime differences, so did some experiments. Looks like the difference is deeper than just OS envrionment variables and sys path.